core/src/main/scala/org/apache/spark/scheduler/LiveListenerBus.scala

```diff
@@ -123,6 +129,23 @@ private[spark] class LiveListenerBus(val sparkContext: SparkContext) extends Spa
       eventLock.release()
     } else {
       onDropEvent(event)
+      droppedEventsCounter.incrementAndGet()
+    }
+    // Don't log too frequently
+    if (System.currentTimeMillis() - lastReportTimestamp >= 60 * 1000) {
+      var droppedEvents = droppedEventsCounter.get
+      while (droppedEvents > 0) {
```

Comment:
I do not quite get the purpose of this while loop. Should we just move forward if `droppedEventsCounter.compareAndSet(droppedEvents, 0)` fail? When compareAndSet fails, there must be another thread that has triggered the log, right?